### PR TITLE
db write and read function

### PR DESCRIPTION
**Pull request by JoshuaMonyei**

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -19,12 +19,18 @@ const io = socketIO(server);
 // Pre-Route middlewares
 require("./src/middlewares/pre_route.middleware")(app);
 
+// Express body parser
+app.use(express.urlencoded({ extended: true }));
```
Comment (eni4sure) [inline server/index.js:23]: register this middleware in the **require("./src/middlewares/pre_route.middleware")(app)** module please

**Review by eni4sure — APPROVED** [review]

**Review by Alao-Abiodun — APPROVED** [review]
Cool

File: server/src/controllers/userProfile.controller.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+const uuid = require('uuid');
+const userSchema = require('../models/User.js');
+const appResponse = require('../utils/response');
+const DatabaseConnection = require('../db/database.helper');
+const Users = new DatabaseConnection('User');
+
+exports.userCreate = async (req, res) => {
```
Comment (NdubuisiJr) [inline server/src/controllers/userProfile.controller.js:7]: Refactor the controller to be a class e.g `class userController {}`

File: server/src/controllers/userProfile.controller.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  body.id = uuid.v4();
+  console.log(body);
+
+  const user = await userSchema.validateAsync(body);
+  const response = await Users.create(user);
+
+  res.status(200).send(appResponse(null, response, true));
+}
+
+exports.getAllUsers = async (req, res) => {
```
Comment (NdubuisiJr) [inline server/src/controllers/userProfile.controller.js:18]: Refactor all the methods/functions in this class to be like the ones in other controllers (for consistency)

File: server/src/db/database.helper.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      organization_id: '612a3a914acf115e685df8e3',
+      collection_name: "chess_room",
+      bulk_write: false,
+      object_id: '',
+      filter: {},
+      payload: {}
+    }
+  }
+
+  create = async (payload) => {
```
Comment (NdubuisiJr) [inline server/src/db/database.helper.js:18]: Refactor the methods/functions in this class to have the same signature (style) as the ones in other (controller) classes. 
e.g
```
async create(payload){
//logic
}
```

File: server/src/routes/index.js [hunk 5]
```diff
@@ -4,6 +4,7 @@ const router = require("express").Router();
 // Custom Modules
 const InfoCtrl = require("../controllers/info.controller");
 const GameCtrl = require("../controllers/game.controller");
+const newUser = require("../controllers/userProfile.controller.js")
```
Comment (NdubuisiJr) [inline server/src/routes/index.js:7]: make the name of this constant consistent with the others e.g `GameCtrl`. yours should be `UserCtrl`

File: server/src/routes/index.js [hunk 6]
```diff
@@ -12,6 +13,13 @@ router.get("/sideBar", InfoCtrl.getSideBarInfo);
 
 router.get("/createGame", GameCtrl.create);
 
+// temporary to test db function
+router.post('/dbWriteTemp', newUser.userCreate);
```
Comment (NdubuisiJr) [inline server/src/routes/index.js:17]: Why did you add 'Temp' to the route name?
If you intend to keep it, remove the Temp

**Review by NdubuisiJr — CHANGES_REQUESTED** [review]

**Comment by NdubuisiJr** [comment]
Looks good now...

**Comment by eni4sure** [comment]
Lemme resolve merge conflict and merge

**Comment by NdubuisiJr** [comment]
You now have conflicts due to other merging that has happened. Update your repo by pull from `upstream` and fix the conflicts

**Comment by eni4sure** [comment]
> You now have conflicts due to other merging that has happened. Update your repo by pull from `upstream` and fix the conflicts

it's a package lock conflict
I'll fix that for him

**Comment by NdubuisiJr** [comment]
Yeah... It needs to be fixed. Github won't let me approve this PR until it is fixed
